File: server/src/main/java/org/codi/catan/model/game/HouseType.java
/*
 * @author the-codinator
 * created on 2020/5/28
 */

package org.codi.catan.model.game;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum HouseType {
    settlement(1, 1),
    city(2, 2);

    private final int resourceMultiplier;
    private final int victoryPoints;
}

File: server-azure/src/model/game/house-type.ts
export enum HouseType {
  settlement = 'settlement',
  city = 'city',
}

export function getResourceMultiplier(type: HouseType): 1 | 2 {
  return type === HouseType.city ? 2 : 1;
}

export function getVictoryPoints(type: HouseType): 1 | 2 {
  return type === HouseType.city ? 2 : 1;
}
